users/admin.py

```diff
@@ -110,6 +110,47 @@ def queryset(self, request, queryset):
 
 @admin.register(User)
 class UserAdmin(admin.ModelAdmin):
+    list_display = [
+        "username",
+        "id",
+        "email",
+        "is_active",
+        "is_spam",
+        "is_bot",
+    ]
+    can_delete = False
+    actions = ["mark_selected_as_spam", "soft_delete_selected", "hard_delete_selected"]
+    search_fields = ["username", "email", "pk"]
+    list_filter = ["is_active", "is_spam", "is_bot"]
+
+    def get_actions(self, request):
+        actions = super().get_actions(request)
+        if "delete_selected" in actions:
+            del actions["delete_selected"]
+        return actions
+
+    def mark_selected_as_spam(self, request, queryset: QuerySet[User]):
+        for user in queryset:
+            user.mark_as_spam()
+
+    def soft_delete_selected(self, request, queryset: QuerySet[User]):
+        for user in queryset:
+            user.soft_delete()
+
+    def hard_delete_selected(self, request, queryset: QuerySet[User]):
+        queryset.delete()
+
+
+# Proxy model for Heavyweight admin view
+class UserHeavyweight(User):
+    class Meta:
+        proxy = True
+        verbose_name = "User (Heavyweight View)"
+        verbose_name_plural = "Users (Heavyweight View)"
```

Comment:
I don't see this proxy mode overrides or change anything in the User model, so it looks to me this acts the same as its parent model. What's the difference?

Reply:
You can't register the same model twice with admin, so you have to make a proxy model. It'll be identical in operation, and actually operate on the same table as the original, but able to be registered separately from the original model.
It's literally just to hack a way to register the same table twice in the admin panel.
If you don't think this is acceptable, we can go about this a different way. I couldn't figure out a way to get this to work without having a migration like this other than to make an alternate admin which is accessed through a separate url like `/admin/heavyweight/` or something.

Let me know if you have better ideas about how to address the ticket.

Reply:
I think it's better to optimise regular User admin page instead. 

I bet it's slow because N+1 OR slow db requests. Make a debug round and modify `get_queryset` to  prefetch/annotate counts for heavy thing we make there

Reply:
Ok that's fine by me. Closing this.